### DISCUSSION TITLE
Publish the data collected by Tasmota

### DISCUSSION THREAD

**Discussion by MarcoLeoncino** [arendst/Tasmota#17991] [Show and tell]
Dear all, 

I have managed to read my Häger smart meter with an 8266 board. I can now read the information locally:

![image](https://user-images.githubusercontent.com/111227897/219956469-a3295e48-42ff-4e94-91b9-a564b98c1d4a.png)

I would like to see the same screen remotely and eventually save the data to a database and make some plots.
Can you please give me some suggestions how to do that? Shall I use MQTT?

Thank you
marco

**Comment by barbudor** [comment]
MQTT might be a starting point but won't do everything

As database, Tasmota has the ability to send data to Influxdb or Prometheus even without using MQTT

You can use Grafana with Influxdb to represent data.
I think that Prometheus has it's own graphical representation

**Comment by MarcoLeoncino** [comment]
Thank you very much for your reply. I can see that Influxdb, Prometheus and Grafana are basically available as trial and then they can be used by a subscription. At the moment I could be already happy if I could see the present data also outside my local network. Basically, I can of course connect to my Tasmota panel only when I am inside my network, but I would like to see the data also from any other position. Is there a guideline to make this in an easy way?

**Reply by barbudor** [reply]
You can run InfluxDB and others in your home on a small computer and use a VPN to get into your home network and see the panels
(do not expose your network to the internet !!! use a VPN)
You don't need anything powerfull. Many people are using RaspberryPi but you may also have an old laptop somewhere on a shelf that could do the job of running a linux operating system (Lubuntu or Debian can be very light and run on old laptops)

Cloud based, there are different solutions out there that you can use for free but with limited capabilities (mostly trials or hobbyist)
But at some point, if you want someone to run a service for you you will have to pay for it

You can have a look at Flespi that I'm using as a cloud based MQTT. Max 20 devices. They have the ability to create panels. Not sure about database.
Some other names that comes to my mind : MyMQTTHub, ThinksSpeak, ThingsBoard, AdafruitIO
But you will quickly run out of the free offering and you'll have to get that VISA/Mastercard out

**Reply by MarcoLeoncino** [reply]
Thank you!

**Reply by pkkrusty** [reply]
Kind of a lot of steps to set up a raspberry pi with influxdb and grafana and then OpenVPN, but step-by-step guides are available on the internet and that makes it way easier. Just note that you probably want to limit your retention period in Influxdb, otherwise it'll default to keeping data forever, which can get big.

**Reply by barbudor** [reply]
> otherwise it'll default to keeping data forever

Is that with InfluxDB v2.x ?
I'm still on v1.18 and the default is 90 days

> Kind of a lot of steps to set up a raspberry pi with influxdb and grafana

Point is that you get nothing for nothing
Either you pick up a commercial platform and you may for that (hosting, maintenance, etc...)
Or you gets your hand dirty to build it yourself
It's always the same question "make or buy ?"

**Reply by pkkrusty** [reply]
Reading influxdb 1.8 [docs](https://docs.influxdata.com/influxdb/v1.8/guides/downsample_and_retain/) online I see:
```
Note: When we created the food_data database in step 1, InfluxDB automatically generated an RP named autogen and set it as the DEFAULT RP for the database. The autogen RP has an infinite retention period. With the query above, the RP two_hours replaces autogen as the DEFAULT RP for the food_data database.
```
so I just made sure I changed my default RP to one_year to avoid any possible headaches.

**Reply by barbudor** [reply]
My bad, I did that config more than a year ago so memory flies :roll_eyes: ....
